src/app/src/components/UserProfile.jsx

```diff
@@ -205,6 +205,10 @@ class UserProfile extends Component {
             </div>
         );
 
+        const contributorIdsString = profile.facilityLists
+            .map(list => list.contributor_id)
+            .join('&contributors=');
```

Comment:
This is a resourceful way of dealing with this issue without updating the API, however if a contributor submits facilities exclusively through the API, they will not have any lists, but they will have associated facilities.

To set up this situation I:

- Registered an confirmed a new account
- Generated an API token on the settings page
-  In a separate browser session logging in as c1@example.com, browsed the new user account in the Django admin and added all the permission groups
<img width="862" alt="Screen Shot 2022-10-03 at 1 35 05 PM" src="https://user-images.githubusercontent.com/17363/193677548-d93089e9-7921-4101-a4d6-eb748bc95e7f.png">
- POSTed a facility

```bash
curl -X POST "http://localhost:8081/api/facilities/" -H  "accept: application/json" -H  "Authorization: Token {TOKEN_VALUE_HERE}" -H  "Content-Type: application/json" -d "{    \"sector_product_type\": \"Apparel\",    \"country\": \"China\",    \"name\": \"Nantong Jackbeanie Headwear Garment Co. Ltd.\",    \"address\": \"No.808,the third industry park,Guoyuan Town,Nantong 226500.\"}"
```

The change below appears to work in this API-only case and in my regression testing I was still able to register a new account and edit the profile of a logged in account.

```diff
diff --git a/src/app/src/components/UserProfile.jsx b/src/app/src/components/UserProfile.jsx
index ff5f18ba..b563edfd 100644
--- a/src/app/src/components/UserProfile.jsx
+++ b/src/app/src/components/UserProfile.jsx
@@ -216,7 +216,7 @@ class UserProfile extends Component {
                 {title}
                 <div>
                     <a
-                        href={`/facilities?contributors=${profile.id}`}
+                        href={`/facilities?contributors=${profile.contributorId}`}
                         rel="noopener noreferrer"
                         style={{
                             backgroundColor: '#FFCF3F',
diff --git a/src/app/src/reducers/ProfileReducer.js b/src/app/src/reducers/ProfileReducer.js
index 71b98a47..57b636d4 100644
--- a/src/app/src/reducers/ProfileReducer.js
+++ b/src/app/src/reducers/ProfileReducer.js
@@ -160,6 +160,7 @@ export default createReducer(
                                 profileSummaryFieldsEnum.facilityLists
                             ],
                     },
+                    contributorId: { $set: payload.contributor_id || null },
                 },
                 fetching: { $set: false },
                 error: { $set: null },
diff --git a/src/django/api/serializers.py b/src/django/api/serializers.py
index 41ceb894..cf8c705e 100644
--- a/src/django/api/serializers.py
+++ b/src/django/api/serializers.py
@@ -267,12 +267,14 @@ class UserProfileSerializer(ModelSerializer):
     facility_lists = SerializerMethodField()
     is_verified = SerializerMethodField()
     embed_config = SerializerMethodField()
+    contributor_id = SerializerMethodField()
 
     class Meta:
         model = User
         fields = ('id', 'name', 'description', 'website', 'contributor_type',
                   'other_contributor_type', 'facility_lists', 'is_verified',
-                  'embed_config')
+                  'embed_config', 'contributor_id')
+        read_only_fields = ('contributor_id',)
 
     def get_name(self, user):
         try:
```

Reply:
I didn't realize the contributor ID was associated directly with the user. Thanks for the suggestion! Updated in e16c22e